dapps/marketplace/src/pages/identity/TelegramAttestation.js

```diff
@@ -117,15 +143,40 @@ class TelegramAttestation extends Component {
 
   renderVerifyCode() {
     const { isMobile } = this.props
-    const { openedLink } = this.state
+    const { openedLink, loading } = this.state
 
     const header = isMobile ? null : (
       <fbt desc="TelegramAttestation.title">Verify Your Telegram Account</fbt>
     )
 
     return (
       <>
-        {this.renderGenerateCode()}
+        <TelegramGenerateCode
+          wallet={this.props.wallet}
+          onComplete={result => {
+            if (!result.success) {
+              this.setState({
+                error: result.reason,
+                loading: false,
+                code: null
+              })
+            }
```

Comment:
I think there should be an `else` here or a `return inside the if(!result.success)`. Since setState is called twice

Reply:
Thanks for catching this @sparrowDom. Have fixed that